lib/elixir/src/elixir_parser.yrl

```diff
@@ -622,6 +631,12 @@ handle_literal(Literal, Token) ->
     false -> Literal
   end.
 
+handle_literal(Literal, Token, ExtraMeta) ->
+  case get(wrap_literals_in_blocks) of
+    true -> {'__block__', [ExtraMeta | meta_from_token(Token)], [Literal]};
+    false -> Literal
+  end.
```

Comment:
This duplicates logic of `handle_literal/2`. My suggestion would be something like this:

```erlang
handle_literal(Literal, Token) ->
  handle_literal(Literal, Token, []).

handle_literal(Literal, Token, ExtraMeta) ->
  case get(wrap_literals_in_blocks) of
    true -> {'__block__', ExtraMeta ++ meta_from_token(Token), [Literal]};
    false -> Literal
  end.
```

and then call as `handle_literal(Literal, Token, [{base, 2}])`.